readme.md

```diff
@@ -1,51 +1,102 @@
-# =nil; Foundation's Cryptography Suite
+# =nil; Foundation
 [![Discord](https://img.shields.io/discord/969303013749579846.svg?logo=discord&style=flat-square)](https://discord.gg/KmTAEjbmM3)
 [![Telegram](https://img.shields.io/badge/Telegram-2CA5E0?style=flat-square&logo=telegram&logoColor=dark)](https://t.me/nilfoundation)
 [![Twitter](https://img.shields.io/twitter/follow/nil_foundation)](https://twitter.com/nil_foundation)
 
-Placeholder repository is a collection of various nil-projects. Check out subfolders for more.
-Supported by [=nil; Foundation](https://nil.foundation)
- 
+This repository is a collection of various nil-projects related to zero-knowledge proof. Check out subfolders for more.
+Supported by [=nil; Foundation](https://nil.foundation).
+
 ## Contents
-1. [Dependencies](#Dependencies)
-2. [Build](#Build)
-3. [Licence](#Licence)
+1. [Structure](#structure)
+2. [Dependencies](#dependencies)
+3. [Build & test](#build_&_test)
+4. [Contributing](#contributing)
+5. [Community](#community)
+6. [Licence](#Licence)
 
-### Dependencies
+## Structure
+root
+├── crypto3
+├── debug-tools
+├── parallel-crypto3
+├── proof-producer
 
-Install nix using the following command:
+## Dependencies
+- [clang](https://clang.llvm.org/) (>= 11.0)/GCC (>= 10.0)/MSVC (>= 14.20)
+- [cmake](https://cmake.org) (>= 3.6)
+- [boost](https://boost.org) (>= 1.76)
 
-```
+All demendencies managed by `nix`.
+
+So first install nix using the following command:
+
+```bash
 curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix | sh -s -- install
 ```
 
-### Build
+then allow `nix-command` and `flakes` experimental features by adding line
 
-For most cases, you want to have an incremental build:
-```
-nix develop .#crypto3-debug-tests
-eval "$configurePhase" // automatically move to the build directory
-eval "$buildPhase"
-eval "$checkPhase"
+```bash
+experimental-features = nix-command flakes
 ```
 
-if you want to build a single test:
+in nix configuration file (`/etc/nix/nix.conf`).
+
+## Build & test
+To activate Nix development environment:
+
+```bash
+nix develop
 ```
-nix develop .#crypto3-debug-tests
-eval "$configurePhase" // automatically move to the build directory
-ninja TEST_NAME
+
+To run all tests:
+
+```bash
+nix flake check
 ```
 
-To build and test an individual project (crypto3, for example):
+To build an individual target:
+```bash
+nix build -L .#<target>
```

Comment:
This is the only place where you are using `target` term, in other similar places `derivation` is used. Seems confusing to me, because this is still a derivation.

Reply:
yes, it may confuse